astropy/units/physical.py

```diff
@@ -1,133 +1,461 @@
-# -*- coding: utf-8 -*-
 # Licensed under a 3-clause BSD style license - see LICENSE.rst
 
 """
-Defines physical unit names.
+Defines the physical types that correspond to different units.
 
 This module is not intended for use by user code directly.  Instead,
-the physical unit name of a `Unit` can be obtained using its `ptype`
-property.
+the physical unit name(s) of a `Unit` can be obtained using its
+``physical_type`` property.
 """
 
+import numbers
+import re
+import warnings
 
 from . import core
 from . import si
 from . import astrophys
 from . import cgs
-from . import imperial
 from . import misc
 
+__all__ = ["def_physical_type", "get_physical_type", "PhysicalType"]
 
-__all__ = ['def_physical_type', 'get_physical_type']
-
+_units_and_physical_types = [
+    (core.Unit(1), "dimensionless"),
+    (si.m, "length"),
+    (si.m ** 2, "area"),
+    (si.m ** 3, "volume"),
+    (si.s, "time"),
+    (si.rad, "angle"),
+    (si.sr, "solid angle"),
+    (si.m / si.s, "speed"),
+    (si.m / si.s ** 2, "acceleration"),
+    (si.Hz, "frequency"),
+    (si.g, "mass"),
+    (si.mol, "amount of substance"),
+    (si.K, "temperature"),
+    (si.W * si.m ** -1 * si.K ** -1, "thermal conductivity"),
+    (si.J * si.K ** -1, {"heat capacity", "entropy"}),
+    (si.J * si.K ** -1 * si.kg ** -1, {"specific heat capacity", "specific entropy"}),
+    (si.N, "force"),
+    (si.J, {"energy", "work", "torque"}),
+    (si.J * si.m ** -2 * si.s ** -1, {"energy flux", "irradiance"}),
+    (si.Pa, {"pressure", "energy density", "stress"}),
+    (si.W, {"power", "radiant flux"}),
+    (si.kg * si.m ** -3, "mass density"),
+    (si.m ** 3 / si.kg, "specific volume"),
+    (si.mol / si.m ** 3, "molar concentration"),
+    (si.m ** 3 / si.mol, "molar volume"),
+    (si.kg * si.m / si.s, {"momentum", "impulse"}),
+    (si.kg * si.m ** 2 / si.s, {"angular momentum", "action"}),
+    (si.rad / si.s, "angular speed"),
+    (si.rad / si.s ** 2, "angular acceleration"),
+    (si.rad / si.m, "plate scale"),
+    (si.g / (si.m * si.s), "dynamic viscosity"),
+    (si.m ** 2 / si.s, {"diffusivity", "kinematic viscosity"}),
+    (si.m ** -1, "wavenumber"),
+    (si.m ** -2, "column density"),
+    (si.A, "electrical current"),
+    (si.C, "electrical charge"),
+    (si.V, "electrical potential"),
+    (si.Ohm, {"electrical resistance", "impedance", "reactance"}),
+    (si.Ohm * si.m, "electrical resistivity"),
+    (si.S, "electrical conductance"),
+    (si.S / si.m, "electrical conductivity"),
+    (si.F, "electrical capacitance"),
+    (si.C * si.m, "electrical dipole moment"),
+    (si.A / si.m ** 2, "electrical current density"),
+    (si.V / si.m, "electrical field strength"),
+    (
+        si.C / si.m ** 2,
```

Comment:
Why so weirdly offset? I'd just indent the set by 4 spaces, and keep the closing parenthesis behind it (or break after the set items)

Reply:
Oh...I got in the habit of using [black](https://black.readthedocs.io/en/stable/) for formatting and have gotten accustomed to that style.  I'll change this to match the Astropy style.

Reply:
Yes, I noticed it is taking python programmers by storm...  I'm still a bit on the fence, feeling the urge to minimize line changes is coming at the cost of readability. Though I can see the case for consistency. Anyway, some of the others are quite allergic to it!